patroni/postgresql.py

```diff
@@ -79,6 +81,33 @@ def __init__(self, config):
             self._state = 'running'
             self._role = 'master' if self.is_leader() else 'replica'
 
+    @property
+    def can_rewind(self):
+        """ check if pg_rewind executable is there and that pg_controldata indicates
+            we have either wal_log_hints or checksums turned on
+        """
+        # low-hanging fruit: check if pg_rewind configuration is there
+        if not self.pg_rewind or\
+           not (self.pg_rewind.get('username', '') and self.pg_rewind.get('password', '')):
+            return False
+
+        cmd = ['pg_rewind', '--help']
+        try:
+            ret = subprocess.call(cmd, stdout=open(os.devnull, 'w'), stderr=subprocess.STDOUT)
```

Comment:
stdout=open(os.devnull, 'w')
Hmm, probably this code will lead to FD leak.

Isn't it better to use `subprocess.check_output([], stderr=subprocess.STDOUT)` instead.
It will raise CalledProcessError if the exit code was non-zero.